src/app.ts

```diff
@@ -48,14 +50,34 @@ bot.on("ready", async (): Promise<any> => {
 })
 
 bot.on("interactionCreate", async (ir: Interaction) => {
-  if (!ir.isCommand()) return
-  const command = bot.commands.get(ir.commandName)
-  if (!command) return
+  if (!ir.isCommand()) return;
+  const command = bot.commands.get(ir.commandName);
+  if (!command) return;
+  if (!cooldowns.has(command)) {
+    cooldowns.set(command, new Discord.Collection());
+  }
+
+  const now = Date.now();
+  const timestamps:any = cooldowns.get(command); //Couldn't find declaration type. Marked as 'any'
+  const cooldownAmount = 1 * 60 * 1000;
+
+  if (timestamps.has(ir.user.id)) {
+    const expirationTime = timestamps.get(ir.user.id) + cooldownAmount;
+
+    if (now < expirationTime) {
+      const timeLeft = (expirationTime - now) / 1000;
+      await ir.reply(`Please wait ${timeLeft.toFixed(1)} more second(s) before reusing the \`${command}\` command.`);
+      return;
+    }
+  }
+
+  timestamps.set(ir.user.id, now);
+  setTimeout(() => timestamps.delete(ir.user.id), cooldownAmount);
```

Comment:
But this would have issues with scheduling, and if someone is impatient, and the cooldown expired 1 second ago, they'd get annoyed.
Or if a schedule event every 5 seconds, you wait *up* to 5 seconds longer than you should.

Reply:
yes but the idea is that you check for the cooldown _before_ you execute the command.
for example if the cooldown expired 1 second ago and a user used the command, what would happen is:
1- check if the cooldown expired or not. It has expired 1 second ago, so remove it
2- execute the command
it's the same functionality but without the intensity/inconsistency of using a timeout